lib/factorial.rb

```diff
@@ -1,6 +1,18 @@
 # Computes factorial of the input number and returns it
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: Linear/O(n), where n is the number of digits being multiplied
+# Space complexity: Constant/O(1), to account for the values of the integers
+
 def factorial(number)
-  raise NotImplementedError
+  value = 1
+  
+  if number != nil
+    while number > 0
+      value *= number
+      number -= 1
+    end
+
+    return value
+  else
+    raise ArgumentError, "nil object is not an integer"
```

Comment:
Consider having your exception handling higher up in your method. It's generally best practice to exit early if the input is incorrect.